### ISSUE TITLE
More questions for Machine learning

### ISSUE THREAD

**Issue by chilcot** [Ebazhanov/in-quiz-questions#360]
ML

Your company wants you to build an internal email text-prediction model to speed up the time that employees spend writing emails. What should you do? 

[ ] Include training email data from seasoned employees. 

[ ] Include training email data from employees who write the majority of internal emails. 

[ ] Include training email data from all employees. 

[ ] Include training email data from new employees.

What is this diagram a good example of? 

[ ] a K-means cluster
[ ] K-nearest neighbor 
[ ] a decision tree 
[ ] a linear regression

You work for a large credit card processing company that wants to create targeted promotions for its customers. The data science team created a machine learning system that groups together customers who made similar purchases, and divides those customers based on customer loyalty. How would you describe this machine learning approach? 

[ ] It uses unsupervised learning to cluster together transactions and unsupervised learning to classify the customers. 

[ ] It uses only unsupervised machine learning. 

[ ] It uses supervised learning to create clusters and unsupervised learning for classification. 

[ ] It uses reinforcement learning to classify the customers.

You are using K-nearest neighbor and you have a K of 1. What are you likely to see when you train the model? 

[ ] high variance and low bias 
[ ] low bias and low variance 
[ ] low variance and high bias 
[ ] high bias and high variance

What is stacking? 

[ ] You use different versions of machine learning algorithms. 
[ ] The predictions of one model become the inputs to another. 
[ ] You use several machine learning algorithms to boost your results. [ ] You stack your training set and testing set together.

Are data model bias and variance a challenge with unsupervised learning? 

[ ] No, data model bias and variance are only a challenge with reinforcement learning. 
[ ] Yes, data model bias is a challenge when the machine creates clusters. 
[ ] Yes, data model variance trains the unsupervised machine learning algorithm. 
[ ] No, data model bias and variance involve supervised learning.

You work for an ice cream shop and created the chart below, which shows the relationship between the outside temperature and ice cream sales. What is the best description of this chart? 

[ ] It is a supervised trendline chart. 
[ ] It is a clustering trend chart. 
[ ] It is a linear regression chart. 
[ ] It is a decision tree.

How is machine learning related to artificial intelligence? 

[ ] Artificial intelligence is a form of unsupervised machine learning. 
[ ] Artificial intelligence focuses on classification, while machine learning is about clustering data. 
[ ] Machine learning is a type of artificial intelligence that relies on learning through data. 
[ ] Machine learning and artificial intelligence are the same thing.

Which choice is best for binary classification? 

[ ] K-means 
[ ] logistic regression 
[ ] linear regression 
[ ] Principal Component Analysis (PCA)

Your organization allows people to create online professional profiles. A key feature is the ability to create clusters of people who are professionally connected to one another. What type of machine learning method is used to create these clusters? 

[ ] reinforcement learning 
[ ] supervised machine learning 
[ ] unsupervised machine learning
[ ] binary classification

Your university wants to use machine learning algorithms to help sort through incoming student applications. An administrator asks if the admissions decisions might be biased against any particular group, such as women. What would be the best answer? 

[ ] Machine learning algorithms are powerful enough to eliminate bias from the data. 

[ ] Machine learning algorithms are based on math and statistics, and so by definition will be unbiased. 

[ ] There is no way to identify bias in the data. 

[ ] All human-created data is biased, and data scientists need to account for that.

With traditional programming, the programmer typically inputs commands. With machine learning, the programmer inputs 

[ ] supervised learning 
[ ] data 
[ ] unsupervised learning 
[ ] algorithms

Random forest is a modified and improved version of which earlier technique? 

[ ] boosted trees 
[ ] stacked trees 
[ ] bagged trees 
[ ] aggregated trees

Why is it important for machine learning algorithms to have access to high-quality data? 

[ ] It will take too long for programmers to scrub poor data. 
[ ] If the data is high quality, the algorithms will be easier to develop. 
[ ] Low-quality data requires much more processing power than high-quality data. 
[ ] If the data is low quality, you will get inaccurate results.

Which statement about K-means clustering is true? 

[ ] In K-means clustering, the initial centroids are sometimes randomly selected. 
[ ] K-means clustering is often used in supervised machine learning. 
[ ] To be accurate, you want your centroids outside of the cluster. 
[ ] The number of clusters are always randomly selected.

In K-nearest neighbor, the closer you are to neighbor, the more likely you are to 

[ ] share common characteristics 
[ ] be part of the root node 
[ ] have a Euclidean connection 
[ ] be part of the same cluster

Are data model bias and variance a challenge with unsupervised learning? 

[ ] Yes, data model bias is a challenge when the machine creates clusters. 
[ ] No, data model bias and variance are only a challenge with reinforcement learning. 
[ ] No, data model bias and variance involve supervised learning. 
[ ] Yes, data model variance trains the unsupervised machine learning algorithm.

In the HBO show Si'ffcon Valley, one of the characters creates a mobile application called Not Hot Dog. It works by having the user take a photograph of food with their mobile device. Then the app says whether the food is a hot dog. To create the app, the software developer uploaded hundreds of thousands of pictures of hot dogs. How would you describe this type of machine learning?

[ ] reinforcement machine learning 
unsupervised machine learning 
supervised machine learnin9 
[ ] semi-supervised machine learning

You work for a large pharmaceutical company whose data science team wants to use unsupervised learning machine algorithms to help discover new drugs. What is an advantage to this approach? 

[ ] You will be able to prioritize different classes of drugs, such as antibiotics. 

[ ] You can create a training set of drugs you would like to discover. 

[ ] The algorithms will cluster together drugs that have similar traits. 

[ ] Human experts can create classes of drugs to help guide discovery.

In 2[ ]15, Google created a machine learning system that could beat a human in the game of Go. This extremely complex game is thought to have more gameplay possibilities than there are atoms of the universe. The first version of the system won by observing hundreds of thousands of hours of human gameplay; the second version learned how to play by getting rewards while playing against itself. How would you describe this transition to different machine learning approaches? 

[ ] The system went from from supervised learning to reinforcement learning. 

[ ] The system evolved from supervised learning to unsupervised learning. 

The system evolved from unsupervised learnin9 to supervised learnin9. 

[ ] The system evolved from reinforcement learning to unsupervised learning.

The security company you work for is thinking about adding machine learning algorithms to their computer network threat detection appliance. What is one advantage of using machine learning? 

[ ] It could better protect against undiscovered threats. 

[ ] It would very likely lower the hardware requirements. 

[ ] It would substantially shorten your development time. 

[ ] It would increase the speed of the appliance.

Your organization allows people to create online professional profiles. A key feature is the ability to create clusters of people who are professionally connected to one another. What type of machine learning method is used to create these clusters? 

[ ] binary classification 

[ ] reinforcement learning 

[ ] supervised machine learning 

[ ] unsupervised machine learning

You work for a music streaming service and want to use supervised machine learning to classify music into different genres. Your service has collected thousands of songs in each genre, and you used this as your training data. Now you pull out a small random subset of all the songs in your service. What is this subset called? 

[ ] test data 
[ ] supervised set 
[ ] data cluster 
[ ] big data

How do machine learning algorithms make more precise predictions? 

[ ] Machine learning servers can host larger databases. 
[ ] The algorithms can run on unstructured data. 
[ ] The algorithms are better at seeing patterns in the data. 
[ ] The algorithms are typically run on more powerful servers.

**Comment by Ebazhanov** [comment]
@chilcot forgot to thank you for this issue and attached questions
FYI I moved everything already into repository 
 :point_right: [machine-learning-quiz.md](https://github.com/Ebazhanov/in-quiz-questions/blob/master/machine-learning/machine-learning-quiz.md)
